### PR TITLE
Add missing displayName to PageHeader.TitleArea and PageHeader.ParentLink

### PR DESCRIPTION
**Pull request by lumaxis**
#### Changed

Added `displayName` to `PageHeader.TitleArea` and `PageHeader.ParentLink` sub components.

### Rollout strategy

- [ ] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [x] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/PageHeader/PageHeader.tsx [hunk 1]
```diff
@@ -337,6 +337,7 @@ const TitleArea = React.forwardRef<HTMLDivElement, React.PropsWithChildren<Title
     )
   },
 ) as PolymorphicForwardRefComponent<'div', TitleAreaProps>
+TitleArea.displayName = 'TitleArea'
```
Comment (broccolinisoup) [inline packages/react/src/PageHeader/PageHeader.tsx:340]: This is great, thanks for adding this! We initially had all sub components FC so I think React was able to infer their names but then we refactored a couple if them. 

Do you mind adding the same thing to Parent Link? 🙏🏻

File: .changeset/funny-dots-yawn.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@primer/react': patch
```
Comment (broccolinisoup) [inline .changeset/funny-dots-yawn.md:2]: I don't think we need a changeset for it. We don't include changeset for docs and tests or changes that only impacts dev env. Do you have any concern with that?

If not, we will add `skip changeset` label and checks should be all green after that yeah let me know if you have any concern 🙌🏻

**Review by broccolinisoup — APPROVED** [review]
Thanks so much for pushing this PR 🙏🏻 Left two questions 👀  Approving in advance!

File: packages/react/src/PageHeader/PageHeader.tsx (hunk 1 above)
Comment (lumaxis) [reply packages/react/src/PageHeader/PageHeader.tsx:341]: Not at all, updated! Do you think we need it for `Root`, too?

File: .changeset/funny-dots-yawn.md (hunk 2 above)
Comment (lumaxis) [reply .changeset/funny-dots-yawn.md]: Makes sense, wasn't familiar with the conventions y'all use – removed the changeset file 🙂

File: packages/react/src/PageHeader/PageHeader.tsx (hunk 1 above)
Comment (broccolinisoup) [reply packages/react/src/PageHeader/PageHeader.tsx:341]: I think we have a displayName for Root

```
export const PageHeader = Object.assign(Root, {
  ContextArea,
  ParentLink,
  ContextBar,
  TitleArea,
  ContextAreaActions,
  LeadingAction,
  Breadcrumbs,
  LeadingVisual,
  Title,
  TrailingVisual,
  TrailingAction,
  Actions,
  Description,
  Navigation,
})

PageHeader.displayName = 'PageHeader'
```

**Pull request merged**